src/cascade/model/age_time_grid.py

```diff
@@ -0,0 +1,153 @@
+        # Each mulstd is one record.
+        for kind in PriorKindEnum:
+            mulstd_df = pd.DataFrame(dict(
+                age=[nan],
+                time=[nan],
+                idx=list(range(count)),
+            ))
+            mulstd_df = mulstd_df.assign(**{new_col: nan for new_col in columns})
+            self._mulstd[kind.name] = mulstd_df
+
+    @property
+    def mulstd(self):
+        return self._mulstd
+
+    def age_time(self):
+        yield from zip(np.repeat(self.ages, len(self.times)), np.tile(self.times, len(self.ages)))
+
+    def __getitem__(self, age_time):
+        """
+        Args:
+            age_time (float, float): Gets all rows with this (age, time).
+
+        Returns:
+            pd.DataFrame or pd.Series with columns.
+        """
+        try:
+            age, time = age_time
+        except TypeError as te:
+            if "not iterable" in str(te):
+                raise TypeError(f"Index should be two floats for getting, not {age_time}.")
+            else:
+                raise
+        if isinstance(age, slice) or isinstance(time, slice):
+            raise TypeError(f"Cannot get a slice from an AgeTimeGrid.")
+        rows = self.grid.query("age == @age and time == @time")
+        if len(rows) > 0:
+            return rows[self.columns]
+        else:
+            raise KeyError(f"Age {age} and time {time} not found.")
+
+    def __setitem__(self, at_slice, value):
+        """
+        Args:
+            at_slice (slice, slice): What to change, as integer offset into ages and times.
+            value (priors.Prior): The prior to set, containing dictionary of
+                                  parameters.
+        """
+        try:
+            if len(at_slice) != 2:
+                raise ValueError("Set value at an age and time, so two arguments.")
+        except TypeError:
+            raise ValueError("Set value at an age and time, so two arguments")
+        at_range = list()
+        for one_slice in at_slice:
+            if not isinstance(one_slice, slice):
+                one_slice = slice(one_slice, one_slice)
+            if one_slice.step is not None:
+                raise ValueError("Slice in age or time, without a step.")
+            start = one_slice.start if one_slice.start is not None else -inf
+            stop = one_slice.stop if one_slice.stop is not None else inf
+            at_range.append([start - GRID_SNAP_DISTANCE, stop + GRID_SNAP_DISTANCE])
+        ages = self.ages[(at_range[0][0] <= self.ages) & (self.ages <= at_range[0][1])]
+        times = self.times[(at_range[1][0] <= self.times) & (self.times <= at_range[1][1])]
+        if len(ages) == 0:
+            raise ValueError(f"No ages within range {at_range[0]}")
+        if len(times) == 0:
+            raise ValueError(f"No times within range {at_range[1]}")
+        self.grid.loc[np.in1d(self.grid.age, ages) & np.in1d(self.grid.time, times), self.columns] = value
+
+    def __len__(self):
```

Comment:
Is this really a meaningful length for this object? My intuition that "length" on multi-dimensional structures is the length of the first axis but that's not what you calculate here. Maybe this structure just shouldn't have len?

Reply:
Maybe this shouldn't be len. There's one really important length here, the number of model variables. I put a `count()` method on the `DismodGroups` which then adds up the lengths of all contained items. I could make it `variable_count()`?

Reply:
I think I like that better. `len` has ambiguous semantics for complex structures. And I think it's only worth dealing with that if you have to.

Reply:
Let's deal with it now.